python/ray/serve/api.py

```diff
@@ -1008,3 +1017,39 @@ def __call__(self, requests):
     """
     f._serve_accept_batch = True
     return f
+
+
+def ingress(
+        app: Union[FastAPI, APIRouter, None] = None,
+        path_prefix: Optional[str] = None,
+):
+    """Mark a FastAPI application ingress for Serve.
+
+    Args:
+        app(FastAPI,APIRouter,None): the app or router object serve as ingress
+            for this backend.
+        path_prefix(str,None): The path prefix for the ingress. For example,
+            `/api`. Serve uses the deployment name as path_prefix by default.
+
+    Example:
+    >>> app = FastAPI()
+    >>> @serve.deployment
+        @serve.ingress(app)
+        def app():
+            pass
+    >>> app.deploy()
+    """
+
+    def decorator(f):
+        if app is not None:
+            f._serve_asgi_app = app
+        if path_prefix is not None:
+            f._serve_path_prefix = path_prefix
+
+        @wraps(f)
+        def inner(*args, **kwargs):
+            return f(*args, **kwargs)
```

Comment:
Don't we only need to handle classes here? This looks like it only handles functions?